addons/web/static/src/js/views/kanban/search_panel.js

```diff
@@ -158,6 +161,9 @@ var SearchPanel = Widget.extend({
         if (!_.contains(validValues, value)) {
             var storageKey = this._getLocalStorageKey(category);
             value = this.call('local_storage', 'getItem', storageKey);
+            if (!value && this.preselectFirstRecord) {
```

Comment:
I think this logic makes more sense right below, something like this:

```js
// if not set in localStorage, select 'All' or 1st record
category.activeValueId = _.contains(validValues, value) ? value :
   this.options.preselectFirst ? validValues[0] : false;
```

btw, is it intended to get from local storage before the 1st record option? If so, the docstring for this option should tell that stored category in local storage has precedence over this option.

Reply:
I guess that we want the localStorage to take the precedence. If there's nothing valid in the localStorage, then we fallback on the default, which is typically 'All' except if 'preselectFirst' has been set to true.

Reply:
Also, please, don't use nested ternary operations